evoxbench/benchmarks/mosegnas.py

```diff
@@ -0,0 +1,353 @@
+        return [sum(x * w for x, w in zip(inputs, weights_row)) + b for weights_row, b in zip(weights, biases)]
+    
+    def sigmoid(self, x):
+        return 1 / (1 + math.exp(-x))
+    
+    def forward(self, x):
+        # Input layer
+        outputs = self.linear(x, self.weights[0], self.biases[0])
+        outputs = [self.relu(x) for x in outputs]
+
+        # Hidden layers
+        for layer in range(self.n_hidden_layers):
+            outputs = self.linear(outputs, self.weights[layer], self.biases[layer])
+            outputs = [self.relu(x) for x in outputs]
+
+        # Output layer
+        outputs = self.linear(outputs, self.weights[-1], self.biases[-1])
+
+        return outputs
+
+    def train(self):
+        pass
+
+class MoSegNASSurrogateModel(SurrogateModel):
+    def __init__(self,
+                surrogate_pretrained_list = None,
+                pretrained_json = None,
+                lookup_table = None,
+                categories = [list(range(4))] * 24,
+                **kwargs):
+        super().__init__()
+
+        self.pretrained_result = pretrained_json
+        self.lookup_table = lookup_table
+        self.categories = categories
+        self.searchSpace = MoSegNASSearchSpace()
+
+        # 10 model
+        if 'latency' in surrogate_pretrained_list:
+            self.latency_pretrained = surrogate_pretrained_list['latency']
+        if 'mIoU' in surrogate_pretrained_list:
+            self.mIoU_pretrained = surrogate_pretrained_list['mIoU']
+        else: 
+            self.latency_pretrained = None
+            self.mIoU_pretrained = None
+
+
+    def name(self):
+        return 'MoSegNASSurrogateModel'
+
+    def fit(self, subnet):
+        """ subnet = [{'d': [...], 'e': [...], 'w': [...]}]
+            pretrained result = [{'config': {'d': [...], 'e': [...], 'w': [...]}, 'params': ..., 'flops': ..., 'latency': ..., 'FPS': ..., 'mIoU': ...}, {...}, {...} """
+        """ method to perform forward in a surrogate model from data """
+        for result in self.pretrained_result:
+            if 'config' in result and isinstance(result['config'], dict):
+                config = result['config']
+                if all(key in config and config[key] == value for key, value in subnet[0].items()):
+                    return [result['params'], result['flops'], result['latency'], result['mIoU']]
+        return None
+
+    def addup_predictor(self, subnet):
+        """ method to predict performance only for flops or params from given architecture features(subnets) """  
+        lookup_table = json.load(open(self.lookup_table, 'r'))
+        d_len = len(self.searchSpace.depth_list)
+        w_len = len(self.searchSpace.width_list)
+        e_len = len(self.searchSpace.categories) - d_len - w_len
+        
+        depth = subnet[:d_len]
+        expand_ratio = subnet[d_len:d_len + e_len]
+        expand_ratio = [self.searchSpace.expand_ratio_list[i] for i in expand_ratio]
+        width = subnet[e_len+d_len:]
```

Comment:
It's recommended to format your code with `black`.
```suggestion
        width = subnet[e_len + d_len :]
```